libs/constrained-generators/src/Constrained/Test.hs

```diff
@@ -102,6 +102,18 @@ prop_complete s =
     -- or fall into an inifinite loop
     pure $ length (show a) > 0
 
+prop_shrink_sound :: HasSpec fn a => Spec fn a -> Property
+prop_shrink_sound s =
+  QC.forAll (strictGen $ genFromSpec s) $ \ma -> fromGEDiscard $ do
+    a <- ma
+    let shrinks = shrinkWithSpec s a
+    pure $
+      cover 40 (not $ null shrinks) "non-null shrinks" $
+        if null shrinks
+          then property True
+          else QC.forAll (QC.elements shrinks) $ \a' ->
+            conformsToSpecProp a' s
+
```

Comment:
Is there where we add the fact that we must filter by conformsTo.  That works for this Property, but where do we do this for any property that uses shrinking?

Reply:
I'm not sure I understand what you mean?